src/wrappers/themis/themispp/secure_cell.hpp

```diff
@@ -105,23 +115,41 @@ namespace themispp{
       secure_cell_optional_context_t(password){}
 
     const data_t& encrypt(data_t::const_iterator data_begin, data_t::const_iterator data_end,  data_t::const_iterator context_begin, data_t::const_iterator context_end){
+      if(data_end-data_begin==0){
```

Comment:
this shouldn't be negative, right?

```suggestion
      if(data_end-data_begin<=0){
```

Reply:
Uh... Please stop reminding me why I don't think STL iterators are well-designed! :sob: :laughing: (These things are totally meant to hide the 'unusable, unsafe and leaky' abstraction of raw C pointers.)

You're right, we should also check for the case when someone swaps the _begin_ and _end_ iterators accidentally. Otherwise we cast the negative difference as `size_t` and end up with a giant number.

(Well, there's still a case when these are iterators of two different containers, but well... we can't do anything about that. Debug builds of STL should probably catch these errors though.)

Reply:
well, i'm just trying to "fuzz" possibilities :)